backend/apid/graphql/entity.go

```diff
@@ -58,6 +68,38 @@ func (r *entityImpl) Author(p graphql.ResolveParams) (interface{}, error) {
 	return handleControllerResults(user, err)
 }
 
+// Related implements response to request for 'related' field.
+func (r *entityImpl) Related(p schema.EntityRelatedFieldResolverParams) (interface{}, error) {
+	entity := p.Source.(*types.Entity)
+
+	ctx := types.SetContextFromResource(p.Context, entity)
+	entities, err := r.entityCtrl.Query(ctx)
+	if err != nil {
+		return []*types.Entity{}, err
+	}
+
+	scores := map[int]int{}
+	for i, en := range entities {
+		matched := strings.Intersect(
+			append(en.Subscriptions, en.Class, en.System.Platform),
+			append(entity.Subscriptions, entity.Class, entity.System.Platform),
+		)
+		scores[i] = len(matched)
+	}
+	sort.Slice(entities, func(i, j int) bool {
+		return scores[i] < scores[j]
+	})
+
+	limit := p.Args.Limit
+	if limit > len(entities) {
+		limit = len(entities)
+	} else if limit < 0 {
+		limit = 0
+	}
```

Comment:
Good place to use the `constrainInt` function from #1297.

Reply:
yup!